packages/backend/src/nest/storage/communityMetadata/communityMetadata.store.ts

```diff
@@ -0,0 +1,201 @@
+      logger('Replicated community metadata')
+      emitter.emit(StorageEvents.COMMUNITY_METADATA_LOADED, await this.getCommunityMetadata())
+    })
+
+    // @ts-expect-error - OrbitDB's type declaration of `load` lacks 'options'
+    await this.store.load({ fetchEntryTimeout: 15000 })
+  }
+
+  public getAddress() {
+    return this.store?.address
+  }
+
+  public async close() {
+    await this.store?.close()
+  }
+
+  public async updateCommunityMetadata(newMeta: CommunityMetadata): Promise<CommunityMetadata | undefined> {
+    try {
+      if (!CommunityMetadataStore.validateCommunityMetadata(newMeta)) {
+        // TODO: Send validation errors to frontend or replicate
+        // validation on frontend?
+        logger.error('Failed to update community metadata')
+        return
+      }
+
+      logger(`About to update community metadata`)
+      if (!newMeta.id) return
+
+      // FIXME: update community metadata if it has changed (so that
+      // we can migrate community metadata easily)
+      const oldMeta = this.store.get(newMeta.id)
+      if (oldMeta?.ownerCertificate && oldMeta?.rootCa) {
+        return oldMeta
+      }
+
+      logger(`Updating community metadata`)
+      // @ts-expect-error - OrbitDB's type declaration of OrbitDB lacks identity
+      const orbitDbIdentity = this.orbitDb.identity.id
+      const meta = {
+        ...oldMeta,
+        ...newMeta,
+        orbitDbIdentity,
+      }
+      await this.store.put(meta.id, meta)
+
+      this.localDbService.putOwnerOrbitDbIdentity(orbitDbIdentity)
+
+      return meta
+
+    } catch (err) {
+      logger.error('Failed to add user profile', err)
+    }
+  }
+
+  public static async validateCommunityMetadata(communityMetadata: CommunityMetadata): Promise<boolean> {
+    // FIXME: Add additional validation to verify communityMetadata
+    // contains required fields.
+
+    try {
+      const crypto = getCrypto()
+      if (!crypto) {
+        throw new NoCryptoEngineError()
+      }
+
+      // TODO: Should we sign community metadata with root private key
+      // and verify? I'm not sure it matters that much.
+
+      const rootCert = loadCertificate(communityMetadata.rootCa)
+      const ownerCert = loadCertificate(communityMetadata.ownerCertificate)
+
+      // Verify that owner certificate is signed by root certificate
+      return await ownerCert.verify(rootCert)
+
+    } catch (err) {
+      logger.error('Failed to validate community metadata:', communityMetadata.id, err?.message)
+      return false
+    }
+  }
+
+  public static async validateCommunityMetadataEntry(ownerOrbitDbIdentity: string, identityProvider: typeof IdentityProvider, entry: LogEntry<CommunityMetadata>): Promise<boolean> {
+    try {
+      if (entry.payload.key !== entry.payload.value.id) {
+        logger.error('Failed to verify community metadata entry:', entry.hash, 'entry key != payload id')
+        return false
+      }
+
+      if (entry.identity.id !== ownerOrbitDbIdentity) {
```

Comment:
I wonder if we can extract some of the validation methods and make them accessible from within wider scope. I mean, we'll be checking data against being put in the db by the community owner not only here but also in certificates store, and probably in even more places in the future.

Reply:
Yea, good idea. I think we might be able to. We can sort of see how patterns play out and then refactor or anticipate certain things and create functions now.

Reply:
You can check the patterns by looking at my PR https://github.com/TryQuiet/quiet/pull/2072